lib/grape/validations.rb

```diff
@@ -9,22 +9,32 @@
 module Grape
   # Registry to store and locate known Validators.
   module Validations
-    class << self
-      attr_accessor :validators
-    end
+    module_function
 
-    self.validators = {}
+    def validators
+      @validators ||= {}
+    end
 
     # Register a new validator, so it can be used to validate parameters.
     # @param short_name [String] all lower-case, no spaces
     # @param klass [Class] the validator class. Should inherit from
     #   Validations::Base.
-    def self.register_validator(short_name, klass)
+    def register_validator(short_name, klass)
       validators[short_name] = klass
     end
 
-    def self.deregister_validator(short_name)
+    def deregister_validator(short_name)
       validators.delete(short_name)
     end
+
+    # Find a validator and if not found will try to load it
+    def lazy_find_validator(short_name)
+      str_name = short_name.to_s
+      validators.fetch(str_name) do
+        register_validator(str_name, "Grape::Validations::Validators::#{str_name.camelize}Validator".constantize)
```

Comment:
`register_validator` will be called twice for this constant(an additional call will be in `inherited` callback)

instead of it we can use a lookup for the constant like
```ruby
validators.fetch(str_name) do
  Grape::Validations::Validators.const_get("{str_name.camelize}Validator")
end

```